src/transactions/LumenEventReconciler.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/LumenEventReconciler.h"
+#include "ledger/LedgerTxn.h"
+#include "transactions/TransactionUtils.h"
+#include "util/Logging.h"
+#include <numeric>
+
+namespace stellar
+{
+
+namespace
+{
+
+int64_t
+calculateDeltaBalance(
+    std::shared_ptr<InternalLedgerEntry const> const& genCurrent,
+    std::shared_ptr<InternalLedgerEntry const> const& genPrevious)
+{
+    auto type = genCurrent ? genCurrent->type() : genPrevious->type();
+    if (type == InternalLedgerEntryType::LEDGER_ENTRY)
+    {
+        auto const* current = genCurrent ? &genCurrent->ledgerEntry() : nullptr;
+        auto const* previous =
+            genPrevious ? &genPrevious->ledgerEntry() : nullptr;
+
+        releaseAssert(current || previous);
+        auto let = current ? current->data.type() : previous->data.type();
+        switch (let)
+        {
+        case ACCOUNT:
+            return (current ? current->data.account().balance : 0) -
+                   (previous ? previous->data.account().balance : 0);
+        default:
+            break;
+        }
+    }
+    return 0;
+}
+
+}
+
+void
+reconcileEvents(AccountID const& txSourceAccount, Operation const& operation,
+                OperationResult const& result, LedgerTxnDelta const& ltxDelta,
+                OpEventManager& opEventManager)
+{
+    // Should only be called pre protocol 8
+    if (result.tr().type() == INFLATION)
+    {
+        return;
+    }
```

Comment:
The `opResult` probably doesn't need to be passed in since you are just checking the result type is `INFLATION` which can be done with the protocol check outside.

Reply:
Done